examples/fluids/src/turb_spanstats.c

```diff
@@ -178,24 +178,27 @@ PetscErrorCode GetQuadratureCoords(Ceed ceed, DM dm, CeedElemRestriction elem_re
 PetscErrorCode SpanStatsSetupDataCreate(Ceed ceed, User user, CeedData ceed_data, ProblemData *problem, SpanStatsSetupData *stats_data) {
   DM       dm = user->spanstats.dm;
   PetscInt dim;
-  CeedInt  P, Q, num_comp_x, num_comp_stats = user->spanstats.num_comp_stats;
+  CeedInt  num_qpts_child, num_comp_x, num_comp_stats = user->spanstats.num_comp_stats;
   Vec      X_loc;
   PetscFunctionBeginUser;
 
   PetscCall(PetscNew(stats_data));
 
   PetscCall(DMGetDimension(dm, &dim));
-  CeedBasisGetNumQuadraturePoints1D(ceed_data->basis_q, &Q);
-  CeedBasisGetNumNodes1D(ceed_data->basis_q, &P);
-
-  PetscCall(GetRestrictionForDomain(ceed, dm, 0, 0, 0, 0, Q, problem->q_data_size_sur, &(*stats_data)->elem_restr_parent_stats,
+  CeedBasisGetNumQuadraturePoints1D(ceed_data->basis_q, &num_qpts_child);
+  num_qpts_child *= dim;
+  PetscCall(GetRestrictionForDomain(ceed, dm, 0, 0, 0, 0, num_qpts_child, problem->q_data_size_sur, &(*stats_data)->elem_restr_parent_stats,
```

Comment:
> This change would make it the same as previous, though I have no idea why we are using the 1D number of quadrature points? That feels funky

It actually wouldn't be the same as previous since `GetRestrictionForDomain` changed to request the _total_ number of quadrature points per element rather than just the number of 1D quadrature points.

The reason I'm using 1D here is that the "parent" basis functions are for 2D elements, but need to match the quadrature rule of the 3D "child" basis functions. It feels funky because it is; we're explicitly taking advantage of the tensor product nature of the quadrature points to do an $L^2$ projection on the 2D mesh based on the spanwise integral.